BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/TransactionDocs/TransactionDocsDataCoordinator.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  TransactionDocsDataCoordinator.swift
+//
+//  Copyright © 2024 Gini GmbH. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import GiniBankAPILibrary
+/// A protocol that defines methods and properties for managing the state of transaction documents in a photo payment flow.
+/// Conforming types are responsible for tracking, modifying, and handling the state related to attaching documents to a transaction.
+public protocol TransactionDocsDataProtocol: AnyObject {
+
+    /// The view controller responsible for presenting document-related views.
+    var presentingViewController: UIViewController? { get set }
+
+    /// Retrieves the current value of the "Always Attach Documents" setting.
+    /// - Returns: A `Bool` representing whether documents should always be attached to the transaction.
+    func getAlwaysAttachDocsValue() -> Bool
+
+    /// Sets the "Always Attach Documents" setting to a given value.
+    /// - Parameter value: A `Bool` indicating whether documents should always be attached to the transaction.
+    func setAlwaysAttachDocs(_ value: Bool)
+
+    /// Checks if there are any attached transaction documents.
+    /// - Returns: A `Bool` indicating whether there are any attached documents.
+    func hasAttachedDocuments() -> Bool
```

Comment:
[Clarification] should we replace it with transactionDocsIds: [String]? If we need to provide ids of attached documents